Src/DependencyCollector/Shared/ServiceBusDiagnosticsEventHandler.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Implements ServiceBus DiagnosticSource events handling.
+    /// </summary>
+    internal class ServiceBusDiagnosticsEventHandler : IDiagnosticEventHandler
+    {
+        public const string DiagnosticSourceName = "Microsoft.Azure.ServiceBus";
+        private const string StatusPropertyName = "Status";
+        private const string EntityPropertyName = "Entity";
+        private const string EndpointPropertyName = "Endpoint";
+
+        private readonly TelemetryClient telemetryClient;
+
+        // Every fetcher is unique for event payload and particular property into this payload
+        // when we first receive an event and require particular property, we cache fetcher for it
+        // There are just a few (~10) events we can receive and each will a have a few payload fetchers
+        // I.e. this dictionary is quite small, does not grow up after service warm-up and does not require clean up
+        private readonly ConcurrentDictionary<Property, PropertyFetcher> propertyFetchers = new ConcurrentDictionary<Property, PropertyFetcher>();
+
+        internal ServiceBusDiagnosticsEventHandler(TelemetryConfiguration configuration)
+        {
+            this.telemetryClient = new TelemetryClient(configuration);
+        }
+
+        public bool IsEventEnabled(string evnt, object arg1, object arg2)
+        {
+            return !evnt.EndsWith(TelemetryDiagnosticSourceListener.ActivityStartNameSuffix, StringComparison.OrdinalIgnoreCase);
```

Comment:
Is `IgnoreCase` intentional? Elsewhere (e.g. `OnEvent`) the case is significant.